### PR TITLE
Image models

### PR DESCRIPTION
**Pull request by alexnwang**
fdddeebe2c283f27a26c63a32efc5ba0447160c7
* Image MAE models

3d3825342e6f1155a9d3daea1a09fde47c346752
* Image DMAE models

6428a366a4015fab09288c8e395d2fc0b04b44a7
* Made it so all D/MAE-ST models save ckpts every 50/200 epochs.
* And all Image-D/MAE models save every 100/800 epochs.

8318bf0767279df66ccc579ce47bc2e2752065af
* Finetune should only save a checkpoint when it gets a better validation loss, copy that to best_checkpoint.pth and delete the saved one.

86ec5233e78768285b4e07c27c4500207e3750db
* Scripts for the Image-D/MAE models.

7e299b8cc061ffa7143a0475417ac3e82f92d32b
* Added code that removes all checkpoints at end of pretraining.
**Please take a look at this commit in particular -- would be very sad if this is broken!**

### CODE REVIEW

File: src/dmae/main_pretrain.py [hunk 1]
```diff
@@ -346,6 +346,12 @@ def main(args):
     total_time = time.time() - start_time
     total_time_str = str(datetime.timedelta(seconds=int(total_time)))
     print("Training time {}".format(total_time_str))
+    
+    ckpt_dir = os.path.dirname(checkpoint_path)
```
Comment (litesaber15) [inline src/dmae/main_pretrain.py:350]: Why not just store one checkpoint.pth file and overwrite it every epoch (or depending upon improvement in val loss) instead of storing one per epoch?

Comment (litesaber15) [reply src/dmae/main_pretrain.py:350]: This is a little brittle since the training workflow can fail before we reach the delete step.

Comment (alexnwang) [reply src/dmae/main_pretrain.py:350]: Perhaps we should consider keeping it. 
We do not get the validation performance, so we there's no signal to decide on during pretraining. I think it is actually worthwhile to keep some history, I've already significantly cut down on the #ckpts, I'd imagine by an order of magnitude. There have been times when the most recent ckpt is messed up, and i've had to rollback to an older one.
